packages/@glimmer/integration-tests/test/helpers/hash-test.ts

```diff
@@ -193,13 +193,36 @@ class HashTest extends RenderTest {
 
     this.registerComponent('Glimmer', 'FooBar', `{{yield @hash}}`, FooBar);
 
-    assert.throws(() => {
-      this.render(`
-        <FooBar @hash={{hash firstName="Godfrey"}} as |values|>
-          {{values.firstName}} {{values.lastName}}
-        </FooBar>
-      `);
-    }, /You attempted to set the "firstName" value on an object generated using the \(hash\) helper|Assignment to read-only properties is not allowed in strict mode/);
+    this.render(
+      `<FooBar @hash={{hash firstName="Godfrey" lastName="Hietala"}} as |values|>{{values.firstName}} {{values.lastName}}</FooBar>`
+    );
+
+    this.assertHTML('Chad Hietala');
+    this.assertStableRerender();
+  }
+
+  @test
+  'defined hash keys are reset whenever the upstream changes'() {
+    class FooBar extends GlimmerishComponent {
+      constructor(owner: object, args: { hash: Record<string, unknown> }) {
+        super(owner, args);
+        args.hash.name = 'Chad';
+      }
+    }
+
+    this.registerComponent('Glimmer', 'FooBar', `{{yield @hash}}`, FooBar);
+
+    this.render(`<FooBar @hash={{hash name=this.name}} as |values|>{{values.name}}</FooBar>`, {
+      firstName: 'Godfrey',
+    });
```

Comment:
Where does `this.name` (passed to the hash invocation) come from? The value passed as the second arg of `this.render` is `firstName`, should this be:

```suggestion
    this.render(`<FooBar @hash={{hash name=this.firstName}} as |values|>{{values.name}}</FooBar>`, {
      firstName: 'Godfrey',
    });
```

Reply:
Ah yes, this was a mistake